css/css-color/parsing/relative-color-computed.html

```diff
@@ -0,0 +1,462 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>CSS Color 5: Computed value of color using relative color syntax</title>
+<link rel="author" title="Sam Weinig" href="mailto:weinig@apple.com">
+<link rel="help" href="https://drafts.csswg.org/css-color-5/#relative-colors">
+<meta name="assert" content="computed value of color using relative color syntax matches expected values">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/css/support/computed-testcommon.js"></script>
+<style>
+    html {
+        --bg-color: blue;
+        --color: green;
+        --accent: lightseagreen;
+        --mycolor: orchid;
+        --mygray: lch(from var(--mycolor) l 0 h);
+    }
+</style>
+</head>
+<body>
+<div id="target"></div>
+<script>
+    // rgb(from ...)
+
+    // Testing no modifications.
+    test_computed_value(`color`, `rgb(from rebeccapurple r g b)`, `rgb(102, 51, 153)`);
+    test_computed_value(`color`, `rgb(from rebeccapurple r g b / alpha)`, `rgb(102, 51, 153)`);
+    test_computed_value(`color`, `rgb(from rgb(10%, 20%, 30%, 40%) r g b / alpha)`, `rgba(26, 51, 77, 0.4)`);
```

Comment:
Sorry @tabatkins, I don't understand. Are you saying the value `rgba(26, 51, 77, 0.4)` is incorrect? What would the correct value be here? (I would love to update WebKit to be compliant if it is not)

Reply:
10% isn't equal to 26, it's equal to 25.5. Implementations are *allowed to* round to their supported precision (I'm not gonna check, but I think it just has to be *at least* 8 bits), but they're not *required* to - it's perfectly valid to output `25.5` as the channel value here, and has been for several years since we changed the syntax from `<integer>` to `<number>`.

Thus you either need to allow two possible correct outputs, rounded and exact, or stick solely to N*20% values, as they're the only values that are both integer percents and integer perbytes.

Reply:
Thanks, that clears things up. I had not realized that rgb()/rgba() serialization was allowed to use `<number>`. My hunch, based trying to do something like that and breaking things (this was a a while ago though), is that it probably isn't web compatible to serialize with `<number>` components, but there is no reason this test has to raise that.

I'll update to use numbers that map to integers as you suggested.